Fix Bayesian Meta-Analysis 2

sophieberkhout opened the pull request:
Fixes https://github.com/jasp-stats/jasp-issues/issues/1236
New pull request (old one: #28) because there were merge conflicts with #22.

Regarding one of the comments in #28 about this plot: 
![image](https://user-images.githubusercontent.com/49243293/120461728-8c3b2b80-c39a-11eb-90f7-a5faaaaafb1b.png)

I kept it like this. It is not possible to show `RE H1` because it is too long and the text will be cut off. I think `r H1` would not be too confusing, since the BF is also labeled with just the `r` and the `f`.

Kucharssim commented on R/bayesianmetaanalysiscommon.R:
```diff
@@ -415,11 +398,63 @@
+  seqResults$lowerMain[1] <- quantile(priorSamples, probs = 0.025)
+  seqResults$upperMain[1] <- quantile(priorSamples, probs = 0.975)
+  
+  # meta analysis cannot run with only 1 study so it starts with 2
+  # the final result is already in the state, so we do not have to run it again (n-1)
```
nice comment!

Kucharssim commented on R/bayesianmetaanalysiscommon.R:
```diff
@@ -1185,51 +1220,54 @@
   } 
   
+  # Scale the height and width of the plot
+  heightCum <- 100 + nrow(dataset) * 30 
```
I personally prefer to call variable names like `heightCumulative` or `totalHeight` - it's only a couple of extra key strokes and adds clarity to the code. But it's your choice of course whether to change it...

sophieberkhout replied:
I changed it to `heightCumulative` :)

Kucharssim commented on R/bayesianmetaanalysiscommon.R:
```diff
@@ -1546,12 +1584,20 @@
   df <- data.frame(effectSize = meanMain, studyLabels = studyLabels, y = length(meanMain):1)
   
+  if(!options$addPrior) {
+    idx <- which(df$studyLabels == "Prior")
```
Nice solution to the indexing from before!

Kucharssim reviewed (commented):
It seems ok to me, the issues are still fixed and the weird QML issue is gone!

I just have a question regarding this:
>I kept it like this. It is not possible to show RE H1 because it is too long and the text will be cut off. I think r H1 would not be too confusing, since the BF is also labeled with just the r and the f.

Can't we name it H<sub>F</sub> and H<sub>R</sub> or something like that (then BF<sub>FR</sub> makes also more sense)? Also, the legend of the plot shows "Prior H<sub>1</sub>" only, but in your case it should then be called "Prior Random H<sub>1</sub>" I think. Or if you embrace my suggestion, just "Prior H<sub>R</sub>". If we resolve this I am happy to merge this :)

sophieberkhout commented:
> Can't we name it HF and HR or something like that (then BFFR makes also more sense)? Also, the legend of the plot shows "Prior H1" only, but in your case it should then be called "Prior Random H1" I think. Or if you embrace my suggestion, just "Prior HR". If we resolve this I am happy to merge this :)

I discussed this with EJ and Quentin and changed a couple of labels accordingly. We chose to use Hr1 and Hf1 (no subscript since the regular H0 and H1 also get no subscripts in the pizza plot in JASP). But then we also had to change the BF labels to BF<sub>_r1f1_</sub>. And also for the constrained random effects analysis. 
Then, it also needed to be changed accordingly for the sequential plot but only for the RE and CRE analyses, as the model-averaged sequential does not need the extra 1 subscript (very complicated, I know haha). 

Regarding the prior, this is the prior for the random effects as well as the model-averaged model. However, unfortunately, there is no posterior for the model-averaged heterogeneity yet.

I also fixed a bug in the table notes for the constrained random effects analysis.

Kucharssim reviewed (approved):
Ok, this looks good now!
